### PR TITLE
Consumer.OnPartitionEOF -> ConsumeResult.IsPartitionEOF

### PR DESCRIPTION
**Pull request by mhowlett**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,11 +15,14 @@
   - Changed the name of `Confluent.Kafka.Avro` to `Confluent.SchemaRegistry.Serdes` (Schema Registry may support other serialization formats in the future).
   - Added a example demonstrating working with protobuf serialized data.
 - Avro serdes no longer make blocking calls to `ICachedSchemaRegistryClient` - everything is `await`ed.
-- References librdkafka.redist [1.0.0-RC2](https://github.com/edenhill/librdkafka/releases/tag/v1.0.0-RC2)
+- References librdkafka.redist [1.0.0-RC4](https://github.com/edenhill/librdkafka/releases/tag/v1.0.0-RC4)
+  - Note: End of partition notification is now disabled by default.
```
Comment (edenhill) [inline CHANGELOG.md]: Tell 'em how it is enabled

File: examples/Consumer/Program.cs [hunk 2]
```diff
@@ -87,6 +85,16 @@ public static void Run_Consume(string brokerList, List<string> topics, Cancellat
                     try
                     {
                         var consumeResult = consumer.Consume(cancellationToken);
+
+                        if (consumeResult.IsPartitionEOF)
+                        {
+                            Console.WriteLine(
+                                $"Reached end of topic {consumeResult.Topic}, partition {consumeResult.Partition}. " +
+                                $"Next message will be at offset {consumeResult.Offset}.");
```
Comment (edenhill) [inline examples/Consumer/Program.cs]: not necessarily. compaction, failover, etc.
I'd rather just have it say "Reached end of .. partition .. at {offset}"

File: src/Confluent.Kafka/Consumer.cs [hunk 3]
```diff
@@ -331,7 +342,8 @@ public ConsumeResult Consume(TimeSpan timeout)
             return new ConsumeResult
             {
                 TopicPartitionOffset = result.TopicPartitionOffset,
-                Message = new Message(result.Message)
+                Message = result.Message,
+                IsPartitionEOF = result.IsPartitionEOF
```
Comment (edenhill) [inline src/Confluent.Kafka/Consumer.cs]: always false, right?

File: examples/Consumer/Program.cs (hunk 2 above)
Comment (mhowlett) [reply examples/Consumer/Program.cs]: argh. people need help to understand what offset represents, but you have a point.

File: src/Confluent.Kafka/Consumer.cs (hunk 3 above)
Comment (mhowlett) [reply src/Confluent.Kafka/Consumer.cs]: nope, a message or partition event could be returned here.

**Pull request merged**